### DISCUSSION TITLE
Trojan:Win32/Wacatac.H!ml threat on running autoortho.exe

### DISCUSSION THREAD

**Discussion by gs3009** [kubilus1/autoortho#124] [General]
HI

I was running autoortho successfully till yesterday.
Suddenly my links to textures are broken and Win 10 quarantines autoortho.exe with a Trojan:Win32/Wacatac.H!ml virus warning.
Cannot run it even after marking it safe.

Pease help

**Comment by gs3009** [comment]
Would appreciate help in running from python source

**Reply by kubilus1** [reply]
To run from source, you must have python 3.10 installed first.  Then you should just need to execute 'run.bat'.

**Comment by kubilus1** [comment]
Can you not just whitelist the exe in Windows defender?

**Comment by gs3009** [comment]
Hi

Thanks for your reply

I have both python 3.10 and 3.11 installed

I have whitelisted autoortho_win.exe and made an exclusion for it too.

Noting seems to work.

I have used it happily for some months.

The problem started when i installed 0.4.2!

And now none of the versons work as it has marked it as malware!

regards

Gautam

On 08-May-2023, at 8:01 PM, kubilus1 ***@***.***> wrote:


Can you not just whitelist the exe in Windows defender?

**Reply by kubilus1** [reply]
It's almost like with Windows you don't actually own your computer at all.

For running from source having both python3.10 and python3.11 could be an issue.  Try uninstalling 3.11.

**Comment by gs3009** [comment]
Ok thanks

Will try tomorrow

Best

Gautam

On 08-May-2023, at 8:09 PM, kubilus1 ***@***.***> wrote:


It's almost like with Windows you don't actually own your computer at all.

For running from source having both python3.10 and python3.11 could be an issue. Try uninstalling 3.11.

**Comment by gs3009** [comment]
Its was working like a charm. Had no issues till Saturday last.

Suddenly Windows doesn't like your compiled executable.


On 08-May-2023, at 8:09 PM, kubilus1 ***@***.***> wrote:


It's almost like with Windows you don't actually own your computer at all.

For running from source having both python3.10 and python3.11 could be an issue. Try uninstalling 3.11.

**Reply by kubilus1** [reply]
Yeah it's been very very frustrating dealing with the chaotic nature of whatever particular voodoo Windows expects on any particular day of the week.  It's a very developer hostile operating system in a lot of ways.

Really makes me appreciate moving to Linux full time all those years ago.

I may be able to come up with some workarounds, but I don't want to put too much time into it, since they could just turn around and block stuff for no reason.

**Comment by gs3009** [comment]
I would really appreciate any solution you can suggest. Will try the python bit. It’s crashes saying cant install winfspy because of wheel. I have upto-date pip and wheel installed Will try send you log tomorrow Thanks again On 08-May-2023, at 20:18, kubilus1 ***@***.***> wrote:﻿
Yeah it's been very very frustrating dealing with the chaotic nature of whatever particular voodoo Windows expects on any particular day of the week.  It's a very developer hostile operating system in a lot of ways.
Really makes me appreciate moving to Linux full time all those years ago.
I may be able to come up with some workarounds, but I don't want to put too much time into it, since they could just turn around and block stuff for no reason.

**Reply by kubilus1** [reply]
Use python3.10 as recommended.

**Comment by gs3009** [comment]
But the executable is being flagged as malware by a lot of sites Pls see the link I posted On 08-May-2023, at 20:18, kubilus1 ***@***.***> wrote:﻿
Yeah it's been very very frustrating dealing with the chaotic nature of whatever particular voodoo Windows expects on any particular day of the week.  It's a very developer hostile operating system in a lot of ways.
Really makes me appreciate moving to Linux full time all those years ago.
I may be able to come up with some workarounds, but I don't want to put too much time into it, since they could just turn around and block stuff for no reason.

**Reply by kubilus1** [reply]
As I've explained elsewhere, pretty much any python packaged exe is flagged now.

**Comment by gs3009** [comment]
Good morning I uninstalled python 3.11 but the batch file wants to keep going to the 3.11 processes which don’t exist anymore. I also wanted to tell you that the executable runs fine till loading screen and installing scenery. It crashes when I Run it. Defender blocks it though I have allowed access So frustrating!Thanks On 08-May-2023, at 20:43, kubilus1 ***@***.***> wrote:﻿
As I've explained elsewhere, pretty much any python packaged exe is flagged now.

**Comment by gs3009** [comment]
HiPython Source and executable both crash after I press RUN!It’s fine upto the scenery check and install stage what could be the cause? Thanks On 08-May-2023, at 20:01, kubilus1 ***@***.***> wrote:﻿
Can you not just whitelist the exe in Windows defender?

**Comment by gs3009** [comment]
Good morning Finally managed to get it working again!Thanks for your help On 08-May-2023, at 20:43, kubilus1 ***@***.***> wrote:﻿
As I've explained elsewhere, pretty much any python packaged exe is flagged now.

**Reply by lknfly1** [reply]
How did you manage it? I also have a serious alert with this malware which seems dangerous based on what I've read.
So far I'm stick with the previous version. 
This malware seems to be implemented in the exe it's hard to know if it is a false positive which is paradoxal knowing that this malware looks pretty dangerous on systems, right?

**Reply by kubilus1** [reply]
It's a false positive that's related to Defender just flagging any python based all in one program as malware.  Super lazy on their part.

Try just using the zip file a packaged with the latest release instead.  TBH I'm tired of hearing about these malware false positives and may stop packing windows releases going forward.

**Reply by lknfly1** [reply]
Thank you for your fast answer, I'll follow your instructions ;-)

**Reply by kubilus1** [reply]
Yeah let me know if the latest zip package works for you.  https://github.com/kubilus1/autoortho/releases/download/0.4.3-1/autoortho_win.zip

It will need to be manually extracted and you'll have to run the autoortho.exe from that dir, but it ought not to trip up Windows Defender.
